mcp/motors/ec_motors.c

```diff
@@ -1178,13 +1305,68 @@ static void piv_init_resolver(void)
 /**
  * @brief initializes the ethercat heartbeat to make sure we remain connected to a motor controller
  * 
- * @param slave_index index of the particular motor to monitor
+ * @param periph_index index of the particular motor to monitor
  */
-static void ec_init_heartbeat(int slave_index)
+static void ec_init_heartbeat(int periph_index)
 {
-    if (slave_index) {
-        ec_SDOwrite16(slave_index, ECAT_HEARTBEAT_TIME, HEARTBEAT_MS);
-        ec_SDOwrite8(slave_index, ECAT_LIFETIME_FACTOR, LIFETIME_FACTOR_EC);
+    if (periph_index) {
+        ec_SDOwrite16(periph_index, ECAT_HEARTBEAT_TIME, HEARTBEAT_MS);
+        ec_SDOwrite8(periph_index, ECAT_LIFETIME_FACTOR, LIFETIME_FACTOR_EC);
+    }
+}
+
+
+/**
+ * @brief Find a distributed clock source on the bus and configure it to send
+ * out sync pulses.
+ * 
+ * @details There must be only a single distributed clock source on the entire
+ * bus. This should be one of the controllers, since the flight computer has
+ * that pesky operating system messing with its timing. So we choose the first
+ * peripheral that has dc enabled as the SYNC source. Everyone else on the bus
+ * gets the same DC_CYCLE (in nanoseconds) but have their sync disabled. For
+ * BEL-090-030 drives, it was found that this sync step works best if it
+ * happens in PRE-OP state, just before requesting transition to SAFE-OP, but
+ * this behavior may vary between drives, see
+ * https://github.com/OpenEtherCATsociety/SOES/issues/151.
+ */
+static void motor_configure_timing(void)
+{
+    int found_dc_source = 0;
+    ec_configdc();
+    while (ec_iserror()) {
+        blast_err("Error after ec_iserror(), %s", ec_elist2string());
+    }
+    for (int i = 1; i <= ec_periphcount; i++) {
+        // TODO(evanmayer): This code is necessary for running with a
+        // distributed clock, which is more accurate. But programming the
+        // configuration is extremely tricky, so leave in machinery for looking
+        // for a clock source, but run in free-run mode for now.
+        // blast_info("Looking for distributed clock on peripheral %d", i);
+        if (!found_dc_source && ec_periph[i].hasdc) {
+            // ec_dcsync0(i, true, ECAT_DC_CYCLE_NS, ec_periph[i].pdelay);
+            found_dc_source = 1;
+            // blast_info("Distributed clock source is peripheral %d", i);
+        } else {
+            // ec_dcsync0(i, false, ECAT_DC_CYCLE_NS, ec_periph[i].pdelay);
+        }
+        // blast_info("Peripheral %d propagation delay is %d", i, ec_periph[i].pdelay);
+        while (ec_iserror()) {
+            blast_err("Peripheral %i, %s", i, ec_elist2string());
+        }
+
+        // Set drives to run in free-run mode
+        // Ref.
+        // https://infosys.beckhoff.com/english.php?content=../content/1033/ethercatsystem/2469122443.html&id=
+        ec_SDOwrite16(i, 0x1C32, 1, 0);
```

Comment:
some register numbers we have comments for some we don't, if theres a comment for this one I missed just delete this. I just don't quite know what this number is

Reply:
The register and available choices for it are documented in the link provided. I'm trying not to re-document the entire EtherCAT spec in our code where I can avoid it

Reply:
I would argue that links are more likely to deprecate but fine